simde/wasm/simd128.h

```diff
@@ -3579,7 +3579,7 @@ simde_wasm_i8x16_shl (simde_v128_t a, uint32_t count) {
       r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
-      r_.neon_i8 = vshlq_s8(a_.neon_i8, vdupq_n_s8(HEDLEY_STATIC_CAST(int8_t, count)));
+      r_.neon_i8 = vshlq_s8(a_.neon_i8, vdupq_n_s8(HEDLEY_STATIC_CAST(int8_t, count & 7)));
```

Comment:
Are these changes needed for all arm64 platforms, or just MS Windows?

Reply:
Yes, all arm platforms, Wasm standard requires
![image](https://user-images.githubusercontent.com/59640930/217563129-19d3401e-a1e2-417e-bcff-16c3289431b0.png)

https://github.com/WebAssembly/simd/blob/main/proposals/simd/SIMD.md

Reply:
I just found another problem:
We used `count & 7` at line 3582 and line 3590, but `count & 7` != `count % 7`. It passed the test case of WABT because WABT tests count = 8, then `count & 7` and `count % 7` happens equal to 0.

We need to change all  `& 7` to `%7` in this file.

Reply:
Test cases added